crypto/ec/curve25519.c

```diff
@@ -5716,24 +5716,38 @@ ossl_x25519_public_from_private(uint8_t out_public_value[32],
     OPENSSL_cleanse(e, sizeof(e));
 }
 
-void
+int
 ossl_x25519_private_from_ed25519(uint8_t out_private_key[32],
                                  const uint8_t private_key[32])
 {
     uint8_t md[SHA512_DIGEST_LENGTH];
+    const EVP_MD *sha512 = EVP_sha512();
```

Comment:
looks like this is causing a build problem.  Believe it is fixable by converting this call to EVP_MD_fetch(NULL, "SHA512", <properties>);

Reply:
sure, I saw that message in CI, but I don‘t know how to fix it. Thank you for pointing it.